src/encoded/utils.py

```diff
@@ -92,3 +99,102 @@ def get_configuration_value(property_name: str,
             if raise_exception is True:
                 raise Exception(f"Cannot get configuration value for: {property_name}\n{get_error_message(e)}")
     return fallback
+
+
+def get_item_with_testapp(
+    testapp: TestApp,
+    identifier: str,
+    collection: Optional[str] = None,
+    frame: Optional[str] = None,
+    status: Optional[Union[int, List[int]]] = None,
+) -> Dict[str, Any]:
+    """Get item from test app."""
+    add_on = get_frame_add_on(frame)
+    resource_path = get_formatted_resource_path(
+        identifier, collection=collection, add_on=add_on
+    )
+    response = testapp.get(resource_path, status=status)
+    if response.status_int == 301:
+        return response.follow().json
+    return response.json
+
+
+def get_frame_add_on(frame: Union[str, None]) -> str:
```

Comment:
You might want to validate that `frame` is one of `raw`, `object` or `embedded` as to catch errors early